packages/components/src/dropdown/index.js

```diff
@@ -108,7 +108,11 @@ export default function Dropdown( props ) {
 					// This value is used to ensure that the dropdowns
 					// align with the editor header by default.
 					offset={ 13 }
-					anchorRef={ ! hasAnchorRef ? containerRef : undefined }
+					anchorRef={
+						! hasAnchorRef
+							? containerRef?.current?.firstChild // Anchor to the rendered toggle.
+							: undefined
+					}
```

Comment:
This always feels weird to me that the prop is called `anchorRef` but it can also accept DOM elements 😆.